packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -570,7 +663,124 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
     return -files.length;
   }
 
-  Map<String, dynamic> _templateContext({
+  Future<void> _updatePubspec(String projectDir, Map<String, dynamic> templateContext, String pluginClass, String androidIdentifier) async {
```

Comment:
I would recommend pulling this logic into a new class that is separate from the create command. in general the use of string processing here is concerning to me. The pubspec.yaml is already structured data, you should take advantage of that to add the platforms.